src/methods/zonal.jl

```diff
@@ -64,7 +64,13 @@ function zonal(f, x::RasterStack; of, skipmissing=true, kw...)
     # TODO: open currently doesn't work so well for large rasterstacks,
     # we need to fix that before we can go back to this being a single method
     # on `RasterStackOrArray`.
-    _zonal(f, _prepare_for_burning(x), of; skipmissing, kw...)
+    return if length(x) < 32
```

Comment:
`length` is not the number of layers, but length like an array, or how many rows the dataframe would have ;)

You want `length(values(x))` or maybe the simplest for the compiler is actually `lenngth(layerdims(x))`

Reply:
```suggestion
    return if length(layerdims(x)) < 32
```